src/main/java/org/lisoft/lsml/model/item/Weapon.java

```diff
@@ -190,11 +197,17 @@ public WeaponRangeProfile getRangeProfile() {
      * Note that this is different from cooldown which is the time the weapon is unavailable between uses, this is
      * the time between activations of the weapon. In particular this includes the time that it takes to charge
      * a gauss rifle, the burn time of lasers, the volley delay from LRMs etc that is not included in cooldown.
+     * 
+     * ammoweapons and energy weaspons calculate this differently. So this is an overridden method.
```

Comment:
@Sestrand says: the truth is ... it is overridden by all
subclasses... and needs to be.  Is there a better way to define this method
to ensure that all subclasses override it?

Reply:
We can bikeshed about this if you'd like; what I'd like addressed is the clarity and value of the comment. I found that the comment was ambiguous from a language point of view and that it subjectively doesn't provide that much value, at least to me, because my IDE (intelliJ community) already renders the method as overridden. 

I see a few ways we could resolve this while keeping us both happy:

1. Make the class `abstract` and mark this method as `abstract` as well, this forces all inheriting, concrete classes to implement the method and makes it obvious to the reader what you're trying to say in the comment. This comes with the downside that this is a rather intrusive change that may entail changing some more code in unit test were `Weapon` objects might be instantiated directly for some tests etc and would need some manual testing to make sure nothing subtly breaks in the UI. There's no precedent for doing this in LSML and I don't know how the database format will deal with it, it might just be a nothing burger, it might also be a hair loss experience :)
2. Just remove the comment about the method being overridden, in Java one has to assume that this is generally a possibility unless the method is marked `final` but that forbids overriding which in and of itself is a bit of a problem as it prevents the method from being mocked in unit tests, see: https://github.com/mockito/mockito/wiki/FAQ.
3. Reword the comment to call attention to specific overriding methods using for example something like this: `Note that this behaviour is changed by inheriting classes. See also: {@see MissileWeapon#getRawFiringPeriod}, ...`.